utils/pruning.py

```diff
@@ -10,7 +10,7 @@
 if os.getenv("TOKENIZERS_PARALLELISM") is None:
     os.environ["TOKENIZERS_PARALLELISM"] = "false"
 
-encoder = SentenceTransformer("sentence-transformers/all-MiniLM-L6-v2", device="cpu")
+encoder = SentenceTransformer("sentence-transformers/all-MiniLM-L6-v2")
```

Comment:
I was thinking of having a variable at the top of `pruning.py` called `user_device`, and then passing it to everywhere torch device is required. We can just check and set it once instead of checking/setting every place we create a tensor.
```py
# at the top
user_device = "cuda" if torch.cuda.is_available() else "cpu"
encoder = SentenceTransformer("sentence-transformers/all-MiniLM-L6-v2", device=user_device)
...
query_emb = encoder.encode(query, convert_to_tensor=True, device=user_device)
...
column_emb = column_emb.to(user_device)
```
The other benefit is that we can easily override `user_device` in tests to run only on cpu. wdyt?